extensions/cornerstone/src/components/ViewportColorbar/ViewportColorbar.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useEffect, useRef } from 'react';
+import { utilities } from '@cornerstonejs/tools';
+import { useSystem } from '@ohif/core';
+import {
+  ColorbarPositionType,
+  TickPositionType,
+  ColorbarCustomization,
+  TickStyleType,
+  ContainerStyleType,
+} from '../../types/Colorbar';
+import { ColorbarRangeTextPosition } from '@cornerstonejs/tools/utilities/voi/colorbar/enums/ColorbarRangeTextPosition';
+
+const { ViewportColorbar: CornerstoneViewportColorbar } = utilities.voi.colorbar;
+
+type ColorbarProps = {
+  viewportId: string;
+  displaySetInstanceUID: string;
+  colormap?: any;
+  colormaps: any[];
+  activeColormapName: string;
+  volumeId?: string;
+  position: ColorbarPositionType;
+  tickPosition: TickPositionType;
+  tickStyles?: TickStyleType;
+  containerStyles?: ContainerStyleType;
+  viewportElementRef?: React.RefObject<HTMLDivElement>;
+};
+
+/**
+ * ViewportColorbar Component
+ * A React wrapper for the cornerstone ViewportColorbar that adds a close button
+ * positioned appropriately based on the colorbar position.
+ */
+const ViewportColorbar = ({
+  viewportId,
+  displaySetInstanceUID,
+  colormaps,
+  activeColormapName,
+  volumeId,
+  position,
+  tickPosition,
+  tickStyles,
+  viewportElementRef,
+}: ColorbarProps) => {
+  const containerRef = useRef<HTMLDivElement>(null);
+  const { servicesManager } = useSystem();
+  const { customizationService } = servicesManager.services;
+
+  useEffect(() => {
+    if (!containerRef.current || !colormaps || !activeColormapName) {
+      return;
+    }
+
+    const viewportElement =
+      viewportElementRef?.current ||
+      (document.getElementById(`viewport-element-${viewportId}`) as HTMLDivElement);
```

Comment:
Can you encapsulate the name of the elements in an imported function?  I really don't like have details like this leaking out of the viewport.  Even something that returns the HTMLDivElement directly, get getViewportElement(viewportId)